rnd/autogpt_server/Dockerfile

```diff
@@ -24,27 +19,41 @@ ENV POETRY_VERSION=1.8.3 \
     PATH="$POETRY_HOME/bin:$PATH"
 RUN pip3 install poetry
 
-FROM server_base AS server_dependencies
+# Copy and install dependencies
+COPY rnd/autogpt_libs /app/rnd/autogpt_libs
+COPY rnd/autogpt_server/poetry.lock rnd/autogpt_server/pyproject.toml /app/rnd/autogpt_server/
+WORKDIR /app/rnd/autogpt_server
+RUN poetry config virtualenvs.create false \
+    && poetry install --no-interaction --no-ansi
+
+# Generate Prisma client
+COPY rnd/autogpt_server/schema.prisma ./
+RUN poetry config virtualenvs.create false && poetry run prisma generate
+
+FROM python:3.11-slim-buster AS server_dependencies
+
+WORKDIR /app
+
+# Copy only necessary files from builder
+COPY --from=builder /app /app
+COPY --from=builder /usr/local/lib/python3.11 /usr/local/lib/python3.11
+COPY --from=builder /usr/local/bin /usr/local/bin
+
+ENV PATH="/app/.venv/bin:$PATH"
 
-RUN mkdir -p /app/autogpt
-RUN mkdir -p /app/forge
 RUN mkdir -p /app/rnd/autogpt_libs
 RUN mkdir -p /app/rnd/autogpt_server
 
-COPY autogpt /app/autogpt
-COPY forge /app/forge
 COPY rnd/autogpt_libs /app/rnd/autogpt_libs
 
 COPY rnd/autogpt_server/poetry.lock rnd/autogpt_server/pyproject.toml /app/rnd/autogpt_server/
 
 WORKDIR /app/rnd/autogpt_server
 
-RUN poetry install --no-interaction --no-ansi
-
 FROM server_dependencies AS server_prisma
 
 COPY rnd/autogpt_server/schema.prisma ./
-RUN poetry run prisma generate
+RUN poetry config virtualenvs.create false && poetry run prisma generate
```

Comment:
This already happens on line 31. I'd advise to remove the unnecessary separate stages and make this file less complicated.

Reply:
Line 31 is a different image. The complexity is to prevent needing 1GB of build tools in the final image